### ISSUE TITLE
Build failure: `OVMFFull.fd` [nixos-23.11]

### ISSUE THREAD

**Issue by Kiskae** [NixOS/nixpkgs#291898]
### Steps To Reproduce

1. build `nix build nixpkgs/nixos-23.11#OVMFFull.fd`

### Build log

```
@nix { "action": "setPhase", "phase": "unpackPhase" }
Running phase: unpackPhase
unpacking source archive /nix/store/b5fvhhqy8432ddm7lpy3h3bk6c13rzbw-edk2-unvendored-src
source root is edk2-unvendored-src
@nix { "action": "setPhase", "phase": "patchPhase" }
Running phase: patchPhase
'BaseTools' -> '/nix/store/cb5c3sy6r12qv8d2cfp7cji85jw3p7hi-edk2-202402/BaseTools'
cp: cannot create regular file 'OvmfPkg/Csm/Csm16/Csm16.bin': No such file or directory
/nix/store/0fpawyv7ya4h142g3pd59i2xhl763rs9-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
```

### Additional context

Seems to have broken just recently, since it was working before I ran a `nix flake update` today.

### Notify maintainers

@adamcstephens @raitobezarius

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.1-zen1, NixOS, 23.11 (Tapir), 23.11.20240225.5bf1cad`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.1`
 - channels(root): `"nixos-22.11"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by stottm** [comment]
I see the same...

nix build nixpkgs/nixos-23.11#OVMFFull.fd
error: builder for '/nix/store/5rs95532hp54ywdwd8ikjs0w0y1sz5vs-OVMF-202402.drv' failed with exit code 1;
       last 7 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/b5fvhhqy8432ddm7lpy3h3bk6c13rzbw-edk2-unvendored-src
       > source root is edk2-unvendored-src
       > Running phase: patchPhase
       > 'BaseTools' -> '/nix/store/cb5c3sy6r12qv8d2cfp7cji85jw3p7hi-edk2-202402/BaseTools'
       > cp: cannot create regular file 'OvmfPkg/Csm/Csm16/Csm16.bin': No such file or directory
       > /nix/store/0fpawyv7ya4h142g3pd59i2xhl763rs9-stdenv-linux/setup: line 131: pop_var_context: head of shell_variables not a function context
       For full logs, run 'nix log /nix/store/5rs95532hp54ywdwd8ikjs0w0y1sz5vs-OVMF-202402.drv'.

Also when using nixos-rebuild switch

error: builder for '/nix/store/5rs95532hp54ywdwd8ikjs0w0y1sz5vs-OVMF-202402.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/nm7gs38nkwxkflkhm0ancj6nxjaj76wx-qemu-ovmf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/46nynk1dh0v74zaiyqr5l5g3y0wwp1fc-unit-script-libvirtd-config-start.drv' failed to build
error: 1 dependencies of derivation '/nix/store/w403xaqg2wvsvylyq5li91j9cgvsk9gk-unit-libvirtd-config.service.drv' failed to build
building '/nix/store/l00sh50mlvnfgpr09bv21490i03f3lgy-unit-script-suid-sgid-wrappers-start.drv'...
error: 1 dependencies of derivation '/nix/store/kihi7q9m9328pivhnls6f4bxxy8ld7ly-system-units.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0dzwwni2p3826pxya11jkacad0vk0634-etc.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5jm16zm2642cnd8favyw7pk1r3f28ygk-nixos-system-nixos-23.11.4801.b7ee09cf5614.drv' failed to build

I am not using flakes and am on stable 23.11 with only a handful of unstable packages.

### Allow unstable packages when prefixed by 'unstable.packageName'
##  eg. unstable.hyprshot
let
  unstable = import <nixos-unstable> { config = { allowUnfree = true; }; };
in

 environment.systemPackages = with pkgs; [
    unstable.gnupg
    unstable.hyprshot
    unstable.swayidle
    unstable.swaylock-effects
];

Here's my vm.nix

{ config, pkgs, ... }:

{

  # Enable dconf (System Management Tool)
  programs.dconf.enable = true;

  # Add user to libvirtd group (username altered for privacy)
  users.users.myuser.extraGroups = [ "libvirtd" ];

  # Install necessary packages
  environment.systemPackages = with pkgs; [
    virt-manager
    virt-viewer
    spice spice-gtk
    spice-protocol
    win-virtio
    win-spice
  ];

  # Manage the virtualisation services
  virtualisation = {
    libvirtd = {
      enable = true;
      qemu = {
        swtpm.enable = true;
        ovmf.enable = true;
        ovmf.packages = [ pkgs.OVMFFull.fd ]; ## THIS IS PROBLEM ##
      };
    };
    spiceUSBRedirection.enable = true;
  };
  services.spice-vdagentd.enable = true;

}

**Comment by adamcstephens** [comment]
This was broken by https://github.com/NixOS/nixpkgs/pull/291054 as upstream dropped this Csm directory. This didn't affect master/unstable as CSM is no longer enabled by default there.

I'll get a fix in but need to do some extra testing to make sure it actually fixes CSM.

If you don't need CSM (legacy BIOS/MBR boot), you can work around this by overriding to disable it.

```nix
(OVMFFull.override { csmSupport = false; }).fd
```

**Comment by adamcstephens** [comment]
This was fixed by #291957
